packages/@react-aria/utils/src/useSyntheticBlurEvent.ts

```diff
@@ -0,0 +1,204 @@
+    // MutationObserver to watch for the disabled attribute, and dispatch these events ourselves.
+    // For browsers that do, focusout fires before the MutationObserver, so onBlur should not fire twice.
+    if (
+      e.target instanceof HTMLButtonElement ||
+      e.target instanceof HTMLInputElement ||
+      e.target instanceof HTMLTextAreaElement ||
+      e.target instanceof HTMLSelectElement
+    ) {
+
+      stateRef.current.observer = new MutationObserver(() => {
+        if (stateRef.current.isFocused && (target as any).disabled) {
+          stateRef.current.observer.disconnect();
+          target.dispatchEvent(new FocusEvent('blur'));
+          target.dispatchEvent(new FocusEvent('focusout', {bubbles: true}));
+        }
+      });
+
+      stateRef.current.observer.observe(target, {attributes: true, attributeFilter: ['disabled']});
+    }
+
+    // Some browsers do not fire onBlur when the document.activeElement is removed from the dom.
+    // Firefox has had a bug open about it for 13 years https://bugzilla.mozilla.org/show_bug.cgi?id=559561
+    // A Safari bug has been logged for it as well https://bugs.webkit.org/show_bug.cgi?id=243749
+    domNodeRemovedObserver.setTarget(target);
+  }, []);
+}
+
+
+class DOMNodeRemovedObserver {
+  private observer: MutationObserver | null;
+  private target: HTMLElement | null;
+
+  constructor() {
+    // don't make an observer in SSR (MutationObserver is not supported in Node)
+    if (typeof window !== 'undefined') {
+      this.observer = new MutationObserver((mutationList) => {
+        if (this.target) {
+          for (const mutation of mutationList) {
+            for (const node of mutation.removedNodes) {
+              if (node?.contains?.(this.target)) {
+                this.target.dispatchEvent(new FocusEvent('blur'));
+                this.target.dispatchEvent(new FocusEvent('focusout', {bubbles: true}));
+                // early return so we don't look at the rest of the DOM
+                return;
+              }
+            }
+          }
+        }
+      });
+    }
+    this.onBlurHandler = this.onBlurHandler.bind(this);
+  }
+
+  onBlurHandler() {
+    this.target = null;
+    if (this.observer) {
+      this.observer.disconnect();
+    }
+  }
+  setTarget(target: HTMLElement) {
+    if (!this.target && target && this.observer) {
+      this.observer.observe(document, {childList: true, subtree: true, attributes: false, characterData: false});
+    }
+    if (this.target !== target) {
+      if (this.target) {
+        this.target.removeEventListener('focusout', this.onBlurHandler);
+      }
+      target.addEventListener('focusout', this.onBlurHandler, {once: true});
+      this.target = target;
+    }
+  }
+  removeTarget(target) {
+    if (this.target === target) {
+      this.onBlurHandler();
+    }
+    if (target) {
+      target.removeEventListener('focusout', this.onBlurHandler);
+    }
```

Comment:
do we need to reset `this.target` to `null` here? I think if we don't line 184 won't do anything when a new target is set.